blackduck/HubRestApi.py

```diff
@@ -231,7 +248,26 @@ def get_limit_paramstring(self, limit):
 
     def get_apibase(self):
         return self.config['baseurl'] + "/api"
-    
+
+    def reauthenticate(self, *args, **kwargs):
```

Comment:
Sorry, this really isn't a good idea. It would be far better to rerun auth rather than explicitly reinitialising the class.

Reply:
I tried rerunning the call to authenticate in my first attempt. I couldn't get it to work, I'll revisit. 
For my own understanding, what is the concern about reinitializing the class upon token expiration? 

Thanks for taking a look!

Reply:
It may have unexpected side effects down the line. Generally speaking in python the only time you should need to manually call `__init__(...)` is when initializing a parent class i.e. `super().__init__`. I wouldn't enforce this sort of change at the library level.